keymap_drawer/parse/rmk.py

```diff
@@ -0,0 +1,408 @@
+            keys_str = layer_config["keys"].strip()
+            
+            # Split keys by whitespace and filter out empty strings, but preserve function calls
+            raw_keys = keys_str.strip()
+            
+            # Parse keys more carefully to handle function calls like MT(C, LCtrl)
+            key_specs = []
+            i = 0
+            while i < len(raw_keys):
+                if raw_keys[i].isspace():
+                    i += 1
+                    continue
+                    
+                # Start of a new key
+                start = i
+                paren_count = 0
+                
+                # Find the end of this key spec
+                while i < len(raw_keys):
+                    if raw_keys[i] == '(':
+                        paren_count += 1
+                    elif raw_keys[i] == ')':
+                        paren_count -= 1
+                    elif raw_keys[i].isspace() and paren_count == 0:
+                        break
+                    i += 1
+                
+                key_spec = raw_keys[start:i].strip()
+                if key_spec:
+                    key_specs.append(key_spec)
+            
+            # Convert to LayoutKey objects
+            layer_keys = []
+            for i, key_spec in enumerate(key_specs):
+                if key_spec == "__":  # Empty key placeholder
+                    layer_keys.append(LayoutKey())
+                else:
+                    layout_key = self._str_to_key(key_spec, layer_names.index(layer_name), [i])
+                    layer_keys.append(layout_key)
+            
+            layers[layer_name] = layer_keys
+            max_keys = max(max_keys, len(layer_keys))
+        
+        # Pad all layers to have the same number of keys
+        for layer_name in layers:
+            while len(layers[layer_name]) < max_keys:
+                layers[layer_name].append(LayoutKey())
+            
+        return layers
+
+    def _get_combos(self, toml_data: dict, layers: dict[str, list[LayoutKey]]) -> list[ComboSpec]:
+        """Extract combos from the TOML data."""
+        combos = []
+        
+        # Look for combo definitions in behavior.combo.combos
+        behavior = toml_data.get("behavior", {})
+        combo_config = behavior.get("combo", {})
+        combo_list = combo_config.get("combos", [])
+        
+        if not combo_list:
+            return combos
+
+        assert self.layer_names is not None
+
+        # Create a mapping from original key specs to positions
+        # We need to parse the keys properly to handle function calls
+        layer_configs = toml_data.get("layer", [])
+        key_spec_to_positions = {}
+        
+        for layer_config in layer_configs:
+            if "keys" not in layer_config:
+                continue
+            
+            keys_str = layer_config["keys"].strip()
+            
+            # Use the same parsing logic as in _get_layers to handle function calls
```

Comment:
This should be a function if we are repeating the exact thing, but I'd rather most of it be replaced by a `re.findall` (see other comment)